architecture-compose/README.md

```diff
@@ -1 +1,101 @@
-# Architecture-compose
\ No newline at end of file
+## Architecture-compose
+This Android library contains composable functions to work with Kaluga states and androidx navigation.
+
+## Installing
+This library is available on Maven Central. You can import Kaluga Base as follows:
+
+```kotlin
+repositories {
+    // ...
+    mavenCentral()
+}
+// ...
+dependencies {
+    // ...
+    implementation("com.splendo.kaluga.architecture.compose:$kalugaVersion")
+}
+```
+
+## Usage
+###Examples
+
+Create a local view model, save in a local ViewModelStore (to ensure cleanup) and 
+link it to lifecycle callbacks and a hardware back button
+
+```kotlin
+@Composable
+fun ContactDetailsLayout(
+    contactDetails: ContactDetails,
+    navigator: Navigator<ContactDetailsNavigation<*>>
+) {
+    val viewModel = store { 
+        remember {
+            ContactDetailsViewModel(contactDetails)
+        }
+    }
+
+    ViewModelComposable(viewModel) { vm ->
+        linkHwBackButtonNavigation {
+            vm.onBackPressed()
+        }
+        Column {
+            //...
+        }
+    }
+}
+```
+
+Set up NavHost navigation in combination with activity navigation
+
+```kotlin
+// action mappers
+fun routeMapper(action: ContactsNavigation<*>): String = 
+    when (action) {
+        is ContactsNavigation.ContactListNavigation.Close -> BACK_ROUTE
+        is ContactsNavigation.ContactListNavigation.ShowContactDetails -> 
+            action.route(Json.encodeToString(action.bundle!!.get(action.type)))
+        is ContactsNavigation.ContactDetailsNavigation.Close, is ContactsNavigation.ShowContacts -> 
+            route<ContactsNavigation.ShowContacts>()
+        else -> throw IllegalStateException()
+    }
+fun activityMapper(action: ContactsNavigation<*>): NavigationSpec = 
+    when (action) {
+        is ContactsNavigation.ContactDetailsNavigation.CallNumber -> 
+            NavigationSpec.Phone(
+                type = Dial,
+                phoneNumber = action.bundle!!.get(action.type)
+            )
+        else -> throw IllegalStateException()
+    }
+
+// constructing and combining navigators
+val routeNavigator = KalugaNavigatorComposeAdapter(
+    rememberNavController(),
+    ::routeMapper
+)
+val combinedNavigator = remember {
+    CombinedNavigator { action: ContactsNavigation<*> ->
+        when(action) {
+            is ContactsNavigation.ContactDetailsNavigation.CallNumber ->
+                ::activityMapper.toActivityNavigator()
+            is ContactsNavigation.ContactListNavigation.Close,
+            is ContactsNavigation.ContactListNavigation.ShowContactDetails,
+            is ContactsNavigation.ContactDetailsNavigation.Close, 
+            is ContactsNavigation.ShowContacts ->
+                routeNavigator
+        }
+    }
+}.apply { bind() }
```

Comment:
Is it possible to move `.apply { bind() }` directly in  a `CombinedNavigatorWrapper` Composable? So that you don't have to call it by yourself everytime

Reply:
good idea. created `rememberCombinedNavigator` composable